app/repos/insurance.js

```diff
@@ -0,0 +1,56 @@
+const sequelize = require('../config/db')
+const { getInsuranceCompany } = require('../lookups')
+
+const createInsurance = async (id, data, transaction) => {
+  if (!transaction) {
+    return sequelize.transaction((t) => createInsurance(data, t))
+  }
+
+  try {
+    const company = await getInsuranceCompany(data.company, transaction)
+
+    if (!company) {
+      throw new Error(`Company not found: ${data.company}`)
+    }
+
+    const insurance = await sequelize.models.insurance.create({
+      company_id: company.id,
+      renewal_date: data.renewalDate,
+      dog_id: id
+    }, { transaction })
+
+    return insurance
+  } catch (err) {
+    console.error(`Error creating insurance: ${err}`)
+    throw err
+  }
+}
+
+const updateInsurance = async (insurance, data, transaction) => {
+  if (!transaction) {
+    return sequelize.transaction((t) => updateInsurance(insurance, data, t))
+  }
+
+  try {
+    const company = await getInsuranceCompany(data.company)
+
+    if (!company) {
+      throw new Error(`Company not found: ${data.company}`)
+    }
+
+    await insurance.update({
```

Comment:
Does this work since the object is being pulled from the CDO object? I presume it does, just surprised that you can use a child object from the CDO query to update the DB